clconf/template/template_funcs.go

```diff
@@ -281,3 +284,87 @@ func RegexReplace(regex, src, repl string) (string, error) {
 	}
 	return re.ReplaceAllString(src, repl), nil
 }
+
+func sortType(input []string) (string, error) {
+	r := "string"
+	if len(input) > 0 {
+		r = input[0]
+	}
+	if r != "string" && r != "int" {
+		return "", fmt.Errorf("sort: Type '%s' is not supported (only int, string)", r)
+	}
+	return r, nil
+}
+
+// Sort sorts a copy of the input as specified type (string, int, default: string).
+func Sort(v []string, asType ...string) ([]string, error) {
+	sortType, err := sortType(asType)
+	if err != nil {
+		return nil, err
+	}
+
+	sortStrings := true
+	var ints []int
+	if sortType == "int" {
+		sortStrings = false
+		ints = make([]int, len(v))
+		for i, val := range v {
+			var err error
+			ints[i], err = strconv.Atoi(val)
+			if err != nil {
+				return nil, err
+			}
+		}
+		sort.Ints(ints)
+	}
+
+	r := make([]string, len(v))
+	for i, val := range v {
+		switch sortType {
+		case "int":
+			r[i] = fmt.Sprintf("%d", ints[i])
+		default:
+			r[i] = val
+		}
+	}
+	if sortStrings { // Don't sort if we've already sorted as ints
+		sort.Strings(r)
+	}
```

Comment:
I think the secondary check can be avoided with a little refactor:

```golang
	sorted := make([]string, len(v))
	if sortType == "int" {
		var ints []int
		sortStrings = false
		ints = make([]int, len(v))
		for i, val := range v {
			var err error
			ints[i], err = strconv.Atoi(val)
			if err != nil {
				return nil, err
			}
		}
		sort.Ints(ints)
		for i, val := range v {
			sorted[i] = strconv.Itoa(ints[i])
		}
	} else {
		// yes, go copy _appears_ backwards, but its copy(dest, src)
		copy(sorted, v)
		sort.Strings(sorted)
	}

	return sorted
```

Reply:
I actually didn't like how it would error if not every key was an int so I rewrote it to allow mixed types which also made it much cleaner.

Reply:
> didn't like how it would error if not every key was an int

wouldn't error be the _logical_ outcome if you told a list of values to _sort using integer comparison_ and one is not an int?